src/Struct/FindologicProduct.php

```diff
@@ -1129,4 +1137,36 @@ protected function getCatUrlPrefix(): string
 
         return rtrim($path, '/');
     }
+
+    protected function sortAndFilterThumbnailsByWidth(MediaThumbnailCollection $thumbnails): MediaThumbnailCollection
+    {
+        $filteredThumbnails = $thumbnails->filter(static function ($thumbnail) {
+            return $thumbnail->getWidth() >= 600;
+        });
+
+        $filteredThumbnails->sort(function (MediaThumbnailEntity $a, MediaThumbnailEntity $b) {
+            return $a->getWidth() <=> $b->getWidth();
+        });
+
+        return $filteredThumbnails;
+    }
+
+    /**
+     * @param MediaThumbnailEntity|MediaEntity $media
+     */
+    private function setImageUrl(Entity $media): void
+    {
+        $encodedUrl = $this->getEncodedUrl($media->getUrl());
+        if (!Utils::isEmpty($encodedUrl)) {
+            $this->images[] = new Image($encodedUrl);
+        }
+    }
+
+    private function addThumbnailUrl(MediaThumbnailEntity $media): void
```

Comment:
Please change the visibility of this method to `protected`, otherwise plugin extensions can not call them and would need to define it on their own.